test/test_prerun.py

```diff
@@ -11,35 +13,113 @@
 from ansiblelint.constants import INVALID_PREREQUISITES_RC
 from ansiblelint.testing import run_ansible_lint
 
+EXECDIR = os.path.dirname(sys.executable)
+
+Deactivator = Callable[[bool], str]
+DeactivateVenv = Callable[[], Deactivator]
+
+
+@pytest.fixture
+def deactivate_venv(monkeypatch: MonkeyPatch) -> Deactivator:
```

Comment:
I like the use of `monkeypatch` but returning a callable seems like an overengineering: since all the tests call it at the beginning of their execution, this fixture could do this when injected.

Assuming you're worried/don't know how this works with `parametrize` — it's quite simple: just inject a `request` fixture here and access the parameter via `request.param`. And on the test function side, add `indirect=('deactivate_venv', )` argument into the `parametrize()` decorator. That's it!